### PR TITLE
Improvements with extensions

### PR DESCRIPTION
**Pull request by ja0b**
This PR include some string extensions ported from Umbraco essentials

### CODE REVIEW

File: Source/Cogworks.Essentials/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -15,7 +15,9 @@ public static bool HasValue(this string input)
             => !string.IsNullOrWhiteSpace(input);
 
         public static Uri ToUri(this string urlString)
-            => new Uri(urlString);
+            => urlString.HasValue()
```
Comment (aochmann) [inline Source/Cogworks.Essentials/Extensions/StringExtensions.cs:18]: This was done by purpose, uri will throw exception if string null or empty.

File: Source/Cogworks.Essentials/Extensions/StringExtensions.cs [hunk 2]
```diff
@@ -222,5 +224,51 @@ public static string RemoveNewLines(this string input)
                     .Replace("\r", string.Empty)
                     .Replace(Environment.NewLine, string.Empty)
                 : string.Empty;
+
+        public static string GetIpAddressWithoutPort(this string ipAddress)
```
Comment (imranhaidercogworks) [inline Source/Cogworks.Essentials/Extensions/StringExtensions.cs:228]: I'm not sure if these methods belong in the string extension

These seem to be very specific to IP addresses and URLs

Maybe a helper would be better?

**Review by imranhaidercogworks — APPROVED** [review]

File: Source/Cogworks.Essentials/Extensions/StringExtensions.cs (hunk 1 above)
Comment (ja0b) [reply Source/Cogworks.Essentials/Extensions/StringExtensions.cs:18]: uh ??

**Review by aochmann — APPROVED** [review]

File: Source/Cogworks.Essentials/Extensions/StringExtensions.cs (hunk 1 above)
Comment (aochmann) [reply Source/Cogworks.Essentials/Extensions/StringExtensions.cs:18]: Sorry read it wrong on beginning. Looks good. If empty or null we are returning null 👍 all good

**Pull request merged**